src/lib/components/ManagePanel.js

```diff
@@ -55,9 +55,22 @@ export default class ManagePanel extends React.Component {
     return url;
   }
 
+  getDate(time) {
+    if (time === 'next') {
+      return 'Next time';
+    }
+    return moment(time).format('ddd, MMM D') || 'Later'
+  }
+
+  getTime(time) {
+    if (time === 'next') {
+      return 'Firefox opens';
+    }
+    return moment(time).format('[@] ha') || '';
+  }
```

Comment:
Sorry, I don't know where to pin this comment, but the sharp edge case here seems to be managing a "Next time Firefox opens" snoozed tab, when you get back into calendar view.

![firefox_developer_edition_and_feat__handle_snoozing_tabs_until_next_open__by_bwinton_ _pull_request__65_ _bwinton_snoozetabs_and_mozilla_-_calendar_-_week_of_jan_9__2017](https://cloud.githubusercontent.com/assets/557895/21820646/6662d34c-d725-11e6-94e2-87992bbef3f6.png)

![firefox_developer_edition_and_feat__handle_snoozing_tabs_until_next_open__by_bwinton_ _pull_request__65_ _bwinton_snoozetabs_and_mozilla_-_calendar_-_week_of_jan_9__2017](https://cloud.githubusercontent.com/assets/557895/21820670/7c120b2c-d725-11e6-8ab2-c050ddf712f7.png)